core/jobs/playtime_entries.py

```diff
@@ -218,3 +238,56 @@ def _fetch_report_records(self, start: datetime, until: datetime) -> Query:
                 PlaytimeSummary.identifier_id,
             )
         )
+
+    @staticmethod
+    def _isbn_for_identifier(
+        identifier: Optional[Identifier],
+        /,
+        *,
+        default_value: str = "",
+    ) -> str:
+        """Find the strongest ISBN match for the given identifier.
+
+        :param identifier: The identifier to match.
+        :param default_value: The default value to return if the identifier is missing or a match is not found.
+        """
+        if identifier is None:
+            return default_value
+
+        if _is_usable_isbn_identifier(identifier):
+            return cast(str, identifier.identifier)
+
+        # If our identifer is not an ISBN itself, we'll use our Recursive Equivalency
+        # mechanism to find the next best one, if available.
+        db = Session.object_session(identifier)
+        eq_subquery = Identifier.recursively_equivalent_identifier_ids_query(
```

Comment:
This method does not use the RecursiveEquivalencyCache table, but instead uses the on-the-go sql function that is deprecated at the moment.

Reply:
The recursive cache table is simpler in usage (and less demanding).
`_db.query(RecursiveEquivalencyCache).filter(parent_id=identifier.id)`

Reply:
Okay. I switched to the cache. It wasn't much simpler to use (unless I'm doing it wrong, which is quite possible). In a way, it makes the tests less convenient because of the need to refresh the cache. But hopefully the performance will be better.